changed timeline display to tabs and added post tweet feature

HadarRose opened the pull request.

Commit:
changed timeline display to tabs and added post tweet feature

jcorteza commented on src/js/components/PostTweet.js:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    postTweet(){
+        this.setState({
```
If you're calling set state later within the same function, do you need to call it here? If it's necessary to make sure the component has a specific state before calling postTweet then it would be a good idea to call postTweet within the setState callback

HadarRose replied:
this is needed to toggle the 'isLoaded' state which controls whether there is a report or not

jcorteza replied:
Since you're updating isLoaded again later it's sort of pointless to set it initially at the beginning of this method. Please read this for reference: https://stackoverflow.com/questions/48563650/does-react-keep-the-order-for-state-updates/48610973#48610973
However, if you feel it is super necessary, postTweet should be moved into the setState callback

jcorteza commented on src/js/components/PostTweet.js:
```diff
@@ -0,0 +1,78 @@
+        };
+    }
+
+    handleInputChange(event){
```
you could probably pass in the value directly based on the way you are referencing this function in your component

kev-chan replied:
agreed.  This is true given how you are handling the call to handleInputChange

jcorteza commented on src/js/components/PostTweet.js:
```diff
@@ -0,0 +1,78 @@
+                    <button
+                        className='post-button'
+                        disabled={!this.state.message?.length || this.state.message == ''}
+                        onClick={() => this.postTweet()}
```
you can replace `() => this.postTweet()` with `this.postTweet` method references work well in react event handlers. especially if you're not passing an event to your function

HadarRose replied:
this actually would pass the wrong 'this', which causes an error when trying to set the state

jcorteza replied:
There's to ways to control for this. 1) you can turn postTweet into an arrow function 2) you can bind `this`
This is a nitpick though

jcorteza commented on src/js/components/timelinePage.js:
```diff
@@ -1,15 +1,63 @@
+        }
+    }
+
+    get TABS(){
```
does this work? Looks like you're missing a space between `get` and `TABS`
Also it looks like this could be an object stored in a variable instead of a function.

HadarRose replied:
since 'const' doesn't work inside of react components, this is the way I found to define readonly objects.

jcorteza replied:
`const` totally works inside react but you have to put it in the right place. In an earlier PR I recommended you define your variables within the `render()` function. You could do that here too.

kev-chan replied:
^ did you try this?

jcorteza commented on src/styles/index.scss:
```diff
@@ -21,139 +21,222 @@ body {  // needed because of some additional imported stylesheet
+            .char-count{
+                text-align: end;
+                color: #737373;
+                &[data-count="280"]{
```
I like the way you used the data-count attribute to update the styles

kev-chan commented on src/js/components/PostTweet.js:
```diff
@@ -0,0 +1,78 @@
+import React from 'react';
```
why is this class capitalized?  details matter

kev-chan commented on src/js/components/timelinePage.js:
```diff
@@ -1,15 +1,63 @@
+            general: {
+                id: 'general-timeline',
+                title: 'General Timeline',
+                render: () => {return <Timeline key='default'/>}
```
does this need a function?

kev-chan commented on src/styles/index.scss:
```diff
@@ -21,139 +21,222 @@ body {  // needed because of some additional imported stylesheet
+
+    .timeline-container {
+        margin: 0px $timeline-margin 0px;
+        .timeline-inputs{
```
i'd recommend newline to help separate the rules

kev-chan reviewed (changes requested):
- hovering over your post button is causing text to move
- add 10px spacing between the  header and the content

Commit:
Practice optimization and minor bug fixes

fixed hover post button bug, made several small changes to functionality of postTwet and timelinePage

Commit:
renaming and optimization of tabs in timelinePage

kev-chan reviewed (changes requested):
code looks fine but i don't see the extra spacing between the tabs and the content that was asked for

Commit:
additional spacing for tabs
